packages/core/src/oldCookiesMigration.ts

```diff
@@ -0,0 +1,36 @@
+import { CookieCache, getCookie, setCookie } from './cookie'
+import { isValidSessionString, persistSession, SessionState } from './sessionManagement'
+
+export const OLD_RUM_COOKIE_NAME = '_dd_r'
+export const OLD_LOGS_COOKIE_NAME = '_dd_l'
+
+// duplicate values to avoid dependency issues
+export const RUM_SESSION_KEY = 'rum'
+export const LOGS_SESSION_KEY = 'logs'
+
+/**
+ * This migration should remain in the codebase as long as older versions are available/live
+ * to allow older sdk versions to be upgraded to newer versions without compatibility issues.
+ */
+export function tryOldCookiesMigration(sessionCookie: CookieCache) {
+  const sessionString = sessionCookie.get()
+  const rumType = getCookie(OLD_RUM_COOKIE_NAME)
+  const logsType = getCookie(OLD_LOGS_COOKIE_NAME)
+  if (rumType || logsType) {
+    setCookie(OLD_LOGS_COOKIE_NAME, '', 0)
+    setCookie(OLD_RUM_COOKIE_NAME, '', 0)
+  }
+  if (!isValidSessionString(sessionString)) {
+    const session: SessionState = {}
+    if (sessionString) {
+      session.id = sessionString
+    }
+    if (logsType && /^[01]$/.test(logsType)) {
+      session[LOGS_SESSION_KEY] = logsType
+    }
+    if (rumType && /^[012]$/.test(rumType)) {
```

Comment:
Is there a way you could explain what those regex are matching for?

Reply:
It is the allowed session types for logs and rum, ex: for [rum](https://github.com/DataDog/browser-sdk/blob/master/packages/rum/src/rumSession.ts#L12)